citest/json_predicate/indexed_predicate_result.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2015 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Implements a composite ValuePredicateResult using an indexed array."""
+
+from .predicate import (
+    CloneableWithNewSource,
+    PredicateResult)
+
+class IndexedPredicateResult(PredicateResult, CloneableWithNewSource):
```

Comment:
`Indexed` and `Keyed` mean very similar things in my mind -- what is the difference in how these are aggregated? Is the `Index` just a number, or is it meant to hide any identifying information in the aggregation?